server/dbData.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * @fileoverview database access object for data.
+ */
+
+var mongodb = require('mongodb');
+var MongoClient = mongodb.MongoClient;
+var mongoUrl = 'mongodb://localhost:27017/express';
+
+var log = require('./log');
+
+/** @type {dbData} */
+module.exports = dbData;
+
+/**
+ * @constructor
+ */
+function dbData() {}
+
+/**
+ * @private @const {string}
+ */
+dbData.DATA_COLLECTION_ = 'file';
+
+/**
+ * @private @const {string}
+ */
+dbData.PROGRESS_COLLECTION_ = 'uploadProgress';
+
+/**
+ * @private @type {string}
+ */
+dbData.USERNAME_ = 'anonymous';
+
+/**
+ * @typedef {{
+ *   error: string
+ * }}
+ */
+dbData.Error;
+
+/**
+ * Creates a connection to the mongodb.
+ * @param {function(!mongodb.Db, Object=)} callback The callback function.
+ */
+dbData.createConnection = function(callback) {
+  MongoClient.connect(mongoUrl, function(err, db) {
+    callback(db, err);
+  });
+};
+
+/**
+ * Inserts a new file into database.
+ * @param {!mongodb.Db} db The database.
+ * @param {string} path Path to the folder of the file.
+ * @param {string} fileName File name of the file.
+ * @param {!Object} property Properties of the file.
+ * @param {function(Object)} callback Callback function.
+ */
+dbData.insertFile = function(db, path, fileName, property, callback) {
+  var collection = db.collection(dbData.DATA_COLLECTION_);
+  collection.insertOne({
+    fileName: fileName,
+    path: path,
+    property: property,
+    user: dbData.USERNAME_
+  }, function(err, result) {
+    if (err) {
+      log.serverLog(err);
```

Comment:
I think err is a compound object. You may want to log err.message. But log(err) might also work if err has a default toString method. Could you check that if this log is behaving correctly?